common/src/main/java/io/netty/util/ConstantPool.java

```diff
@@ -55,31 +55,23 @@ public T valueOf(Class<?> firstNameComponent, String secondNameComponent) {
      * @param name the name of the {@link Constant}
      */
     public T valueOf(String name) {
-        if (name == null) {
-            throw new NullPointerException("name");
-        }
-
-        if (name.isEmpty()) {
-            throw new IllegalArgumentException("empty name");
-        }
-
-        synchronized (constants) {
-            T c = constants.get(name);
-            if (c == null) {
-                c = newConstant(nextId, name);
-                constants.put(name, c);
-                nextId ++;
+        T c;
+        if (exists(name)) {
+            synchronized (constants) {
```

Comment:
This will introduce a race condition. `exists()` and `constants.get()` should be in a single `synchronized` block.